src/script/common/c_content.cpp

```diff
@@ -1564,6 +1585,127 @@ ToolCapabilities read_tool_capabilities(
 	return toolcap;
 }
 
+/******************************************************************************/
+PointabilityType read_pointability(lua_State *L, int index)
+{
+	if (lua_isboolean(L, index)) {
+		if (lua_toboolean(L, index))
+			return POINTABLE;
+		else
+			return POINTABLE_NOT;
+	} else if (lua_isstring(L, index)) {
+		size_t len;
+		const char* s = lua_tolstring(L, index, &len);
+		if (!strcmp(s, "blocking"))
+			return POINTABLE_BLOCKING;
+	}
+	return POINTABLE;
+}
+
+/******************************************************************************/
+PointingAbilities read_pointabilities(lua_State *L, int index)
+{
+	PointingAbilities pointabilities;
+	
+	lua_getfield(L, index, "nodes");
+	if(lua_istable(L, -1)){
+		int ti = lua_gettop(L);
+		lua_pushnil(L);
+		while(lua_next(L, ti) != 0) {
+			// key at index -2 and value at index -1
+			std::string name = luaL_checkstring(L, -2);
+			
+			// handle groups
+			if(name.find("group:") != std::string::npos) {
```

Comment:
`find` is inappropriate here.
You could use string views, e.g. `std::string_view(name).substr(1,6)=="group:"`.

Reply:
Fixed